### PR TITLE
Fixes #599: Autoselect db and files if site is selected.

### LINKED ISSUE #599: In Cloud IDE choosing a site should automatically download the appropriate database and files too
When working through the "Get Started" page by clicking on the "Copy from the cloud platform" button, customers are asked the following questions:

* Choose a Cloud Platform environment
* Choose a site
* Choose a database

It's a cumbersome user experience to pick the corresponding DB and files, especially from the **giant list of Site Factory databases** (and files). Also, when you're not a developer, this is **very** confusing to know which DB/files correspond to the site you're pulling in.

We should just automate this away from our customers to give them a vastly improved UX. In this instance automating doesn't necessarily mean not asking questions, but at least suggesting the best option.

An ideal workflow would be this (using the database as an example):

* As an IDE user, I select site `T`.
* Instead of returning DBs for sites `A` to `Z`, ACLI returns only the dev, test and prod DBs for specifically for site `T`.
* There is clear indication which env the DB is pulled from so I know which one corresponds to my needs.
* I can thus have a shortlist of candidate DBs to pull from and know which environment I wish to pull my DB from

### PR DESCRIPTION
**Pull request by grasmash**
**Motivation**

Fixes #599 

**Proposed changes**

When a user selects an ACSF site database, save that choice and re-use it for the files prompt.

**Alternatives considered**

**Testing steps**

1. Follow the [contribution guide](https://github.com/acquia/cli/blob/master/CONTRIBUTING.md#building-and-testing) to set up your development environment.
2. Clear the kernel cache to pick up new and changed commands: `./bin/acli ckc`
3. (add specific steps for this pr)

**Merge requirements**
- [x] _Bug_, _enhancement_, or _breaking change_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

**Comment by grasmash** [comment]
@anavarre I've tested this with ACSF but I don't have a an ACE multisite to test agaist.

File: src/Command/Pull/PullCommandBase.php [hunk 1]
```diff
@@ -47,6 +47,11 @@ abstract class PullCommandBase extends CommandBase {
    */
   protected $drushHasActiveDatabaseConnection;
 
+  /**
+   * @var string
+   */
+  private $site;
```
Comment (anavarre) [inline src/Command/Pull/PullCommandBase.php:53]: FYI at Acquia we typically try to refer to an application, not a site. I however understand for the purpose of what we're trying to achieve here (pulling a specific site in the IDE), `site` might work better.

File: src/Command/Pull/PullCommandBase.php [hunk 2]
```diff
@@ -612,22 +619,45 @@ protected function runComposerScripts($output_callback = NULL): void {
...
   }
 
+  /**
+   * @param $environment
+   * @param \Symfony\Component\Console\Input\InputInterface $input
+   *
+   * @return mixed|string
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function determineSite($environment, InputInterface $input) {
```
Comment (anavarre) [inline src/Command/Pull/PullCommandBase.php:629]: Are you actually trying to determine the site (application) or the multisite?

**Review by anavarre — APPROVED** [review]
Confirmed when selecting the (multi)site is automatically detects the corresponding DB. I have tried this against a multisite ACE app. I understand you've already confirmed it works for ACSF. If that's the case, LGTM!

```
 Choose a site [myapp]:
  [0] myapp
  [1] myapp2
    ✔ Copying Drupal's public files from the Cloud Platform
        Connecting to database drupal
                                                                                                                        
 [INFO] Using a database backup that is 10 hours old. Backup #185825056 was created at Tue Sep 21 7:12:09 UTC 2021.     
                                                                                                                        
        You can view your backups here:                                                                                 
        https://cloud.acquia.com/a/environments/18472-0e737068-136c-4b7e-8b6e-3fdf0541b428/databases                    
                                                                                                                        
        To generate a new backup, re-run this command with the --on-demand option.                                      
                                                                                                                        

    ✔ Downloading Drupal database copy from the Cloud Platform
    ✔ Installing Composer dependencies
        }
    ✔ Clearing Drupal caches via Drush
    ✔ Sanitizing database via Drush

 Your database was sanitized via drush sql:sanitize. This has changed all user passwords to randomly generated strings. To log in to your Drupal site, use drush uli
```

**Pull request merged**